src/main/java/com/intel/genomicsdb/GenomicsDBImporter.java

```diff
@@ -731,23 +660,57 @@ static File printLoaderJSONFile(
    *
    * Assume one sample per input GVCF file
    *
-   * @param  variants  Variant Readers objects of the input GVCF files
+   * @param sampleToReaderMap  Variant Readers objects of the input GVCF files
    * @param useSamplesInOrderProvided  If True, do not sort the samples,
    *                                   use the order they appear in
+   * @param validateSampleMap Check i) whether sample names are consistent
+   *                          with headers and ii) feature readers are valid
+   *                          in sampleToReaderMap
    * @param lbRowIdx Smallest row idx which should be imported by this object
    * @return  Mappings of callset (sample) names to TileDB rows
    */
-  static GenomicsDBCallsetsMapProto.CallsetMappingPB generateSortedCallSetMap(
-      final Map<String, FeatureReader<VariantContext>> variants,
+  public static GenomicsDBCallsetsMapProto.CallsetMappingPB generateSortedCallSetMap(
+      final Map<String, FeatureReader<VariantContext>> sampleToReaderMap,
       boolean useSamplesInOrderProvided,
+      boolean validateSampleMap,
       final long lbRowIdx)
   {
-    List<String> sampleNames = new ArrayList<>(variants.size());
-    for (Map.Entry<String, FeatureReader<VariantContext>> v : variants.entrySet()) {
-      VCFHeader h = (VCFHeader) v.getValue().getHeader();
-      sampleNames.add(h.getSampleNamesInOrder().get(0));
+    if (!validateSampleMap) {
+      return GenomicsDBImporter.generateSortedCallSetMap(
+          (List) Arrays.asList(sampleToReaderMap.keySet().toArray()),
```

Comment:
This is a weird raw cast.  I suggest `new ArrayList<>(sampleToReaderMap.keySet())` instead.